### ISSUE TITLE
Controller not emulated as Xbox controller - XOutput doesn't work?

### ISSUE THREAD

**Issue by Darthagnon** [csutorasa/XOutput#51]
*Please forgive me if I have misunderstood the purpose of XOutput.*
As far as I can tell, XOutput is supposed to convert inputs from generic controllers and emulate Xbox controller inputs with them? 
I am using a Tesun USB-706 generic Dualshock2-type USB controller. I have set up all the buttons, triggers and joysticks as the corresponding Xbox controls, but it seems to work only within XOutput - the DirectInput and XInput sensors in the Edit menu light up correctly, but once I start a game, no Xbox controllers are detected. 
This has been tested on Terraria and Vagante, both of which natively support Xbox controllers, with no success. I have tested it on my Win7 laptop (using the default ScpToolkit driver - I'm stuck on Windows PowerShell 2.0 for some reason and cannot find a manual way to install the ViGEm driver, which is a pain and a half to install), with the aforementioned results. 
When my brother tested XOutput with the ViGEm driver on his Win10 laptop (comes with the required PowerShell 5.0), it still didn't work in-game, but on the windows desktop, it glitched out, with options in the Start menu constantly cycling, as if an arrow key was being held down.

TL;DR - Tesun USB-706 generic Dualshock 2 seems to work within XOutput, but does not work at all in-game. Using the ViGEm driver only makes things worse.

Any suggestions?

PS. Is csutorasa's XOutput at all related to Stents-'s Xoutput (see https://github.com/Stents-/XOutput/releases)? I didn't have any luck with Stents-'s version, either - it was impossible to tell how to map the analog sticks, and it froze Win7 when it tried to install the custom driver.

**Comment by Darthagnon** [comment]
I have managed to get XOutput to work, sort of. It seemed to start working of its own accord: though I have tried it several times already, and so has my brother, I clicked "Start" and a driver installation message appeared this last time, and it was recognised as an Xbox remote by Vagante. I also tried it with Broforce (also natively supports Xbox remote); this did not recognise it as an Xbox remote - no buttons nor prompts worked, though I could scroll the menus with the left analog stick. I'm wondering if it's because I have nothing mapped as the Home button (there is no physical home button which I can map, this being a USB DS2 - maybe some games detect it as a non-Xbox remote if not all buttons are available? Dunno, though I'll keep testing, see which games work, which don't. NB I'm still using the SCPToolkit, not ViGEm, which I still haven't managed to install (and it bugged out, as in the OP, on my brother's PC).

**Comment by Darthagnon** [comment]
NB I also installed the Xbox remote driver/software from https://www.microsoft.com/accessories/en-gb/d/xbox-360-controller-for-windows just before this last time that it worked. You may wish to add this as a prerequisite to the Readme? Don't quite get why it's required, as I've used an Xbox remote before with my PC with no problems. But oh, well, installing it must have made XOutput work (only relevant change I made to my system that I remember, besides a couple of restarts)

**Comment by csutorasa** [comment]
Thank you for your investigation, I have added the driver installation to [README.md](https://github.com/csutorasa/XOutput/blob/master/README.md)  how to install section in [this commit](https://github.com/csutorasa/XOutput/commit/70a6f94accafce70d8a3fb61d2ff7e2f9c52f1ae).

**Comment by Darthagnon** [comment]
I just realised, after a more careful reading of the readme, that afaik, I never installed ScpToolkit. Meaning I was trying to run XOutput without:
 - ScpToolkit
 - ViGEm
 - Xbox 360 Controller for Windows v1.2
I guess I must have a) assumed that I already had the Xbox remote drivers from previous usage of an Xbox remote; b) assumed ScpToolkit was not supported at all by XOutput; and c) assumed that XOutput was my average open-source portable program.

Interestingly, this also means I now have XOutput apparently working with *only* the Xbox 360 Controller for Windows v1.2 installed (?!). I still have to confirm that it works with more games, but at least I know more or less what to do to get it working on other computers.
